test/available_delivery_location_types.test.js

```diff
@@ -0,0 +1,36 @@
+// These are hard-wired to what I know about patron types 10 & 78
+describe('AvailableDeliveryLocationTypes', function () {
+  let AvailableDeliveryLocationTypes = require('../lib/available_delivery_location_types.js')
+
+  it('maps patron type 10 to [\'Research\']', function () {
+    let tests = []
+
+    let aTest = [Promise.resolve().then(() => {
+      AvailableDeliveryLocationTypes._getPatronTypeOf = () => Promise.resolve('10')
```

Comment:
Since there's now only one promise chain being run in this test, you don't strictly need to set up your `tests` array in this way fwiw. You could just immediately return `aTest` from `it`

Also, doesn't impact the test, but looks like `aTest` is a one-element promise array, which is pushed to `test`, so ultimately you're calling Promise.all([[aTest]]). Promise.all surprisingly seems to await resolution of promises in the nested array, so doesn't seem to matter, but the depth is unnecessary.